### PR TITLE
refine logs of reshard

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

Others
### PR changes

Others
### Description

refine logs of reshard

There are four types of reshard,
- reshard tensor, called by `shard_tensor()`
- reshard func, called by `paddle.distributed.reshard()`
- reshard input, called inside dygraph api, like `paddle.matmul`, when the inputs not match the dist_attr infered by spmd rules
- reshard output, called inside `loss.backward()`, when the grad outputs not match the dist_attr required 

image
image

Pcard-76459

### CODE REVIEW

File: paddle/phi/api/lib/data_transform.cc [hunk 1]
```diff
@@ -639,7 +639,7 @@ std::string ReshardDebugInfo(
     const phi::distributed::DistTensor& src_tensor,
     const phi::distributed::TensorDistAttr& dist_attr) {
   std::stringstream sstream;
-  sstream << "Reshard from src {Global Shape: " << src_tensor.dims()
+  sstream << "reshard from {Global Shape: " << src_tensor.dims()
```
Comment (GhostScreaming) [inline paddle/phi/api/lib/data_transform.cc:642]: 需要统一大写吗

**Comment by GhostScreaming** [comment]
秋大师，`ReshardApiInputToKernelInput`的LOG输出可以截个图放comment里~

File: paddle/phi/api/lib/data_transform.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/phi/api/lib/data_transform.cc:642]: 这里是统一后的小写

**Review by GhostScreaming — APPROVED** [review]
LGTM

**Review by heavyrain-lzy — APPROVED** [review]
LGTM for generator
